src/Iact1dUnbinnedLkl.cc

```diff
@@ -289,66 +289,77 @@ Int_t Iact1dUnbinnedLkl::InterpretInputString(TString inputString)
-      fNon         = dataSet->GetOnSample()->GetEntries();
-      fNoff        = dataSet->GetOffSample()->GetEntries();
-
-      // extract data
-      Double_t eventOnE,eventOffE;
-      dataSet->SetOnBranchAddress("E",&eventOnE);
-      dataSet->SetOffBranchAddress("E",&eventOffE);
-
-      fOnSample  = new Float_t[fNon];
-      fOffSample = new Float_t[fNoff];
-
-      for(Int_t i=0;i<fNon;i++)
-	{
-	  dataSet->GetOnEntry(i);
-	  fOnSample[i] = TMath::Log10(eventOnE);
-	}
-      for(Int_t i=0;i<fNoff;i++)
-	{
-	  dataSet->GetOffEntry(i);
-	  fOffSample[i] = TMath::Log10(eventOffE);
-	}
+      cout << "Iact1dUnbinnedLkl::InterpretInputString Error: you have specified an unsupported IactEventListIrf format!" << endl;
+    }
+
+  // extract info from file 
+  fEpmin       = dataSet->GetEpmin();
+  fEpmax       = dataSet->GetEpmax();
+  fTrueTau     = fDataTau     = fTau      = dataSet->GetTau();
+  fDataDTau    = fDTau     = TMath::Sqrt(TMath::Power(dataSet->GetDTau(),2)+TMath::Power(relDTauSyst*fTau,2));
+  fDataObsTime = fObsTime  = dataSet->GetObsTime();
+  fNon         = dataSet->GetOnSample()->GetEntries();
+  fNoff        = dataSet->GetOffSample()->GetEntries();
+  
+  // extract data
+  Double_t eventOnE,eventOffE;
+  dataSet->SetOnBranchAddress("E",&eventOnE);
+  dataSet->SetOffBranchAddress("E",&eventOffE);
+ 
+  fOnSample  = new Float_t[fNon];
+  fOffSample = new Float_t[fNoff];
 
-      // extract and copy IRFs
-      if(dataSet->GetHAeff()->GetEntries())
-	if(SetAeff(dataSet->GetHAeff()))
-	  cout << "Iact1dUnbinnedLkl::Iact1dUnbinnedLkl Warning: problems setting AEff histogram" << endl;
-      if(dataSet->GetHAeffOff()->GetEntries())
-	if(SetAeffOff(dataSet->GetHAeffOff()))
-	  cout << "Iact1dUnbinnedLkl::Iact1dUnbinnedLkl Warning: problems setting AEffOff histogram" << endl;      
-      if(dataSet->GetGEreso()->GetN() && dataSet->GetGEbias()->GetN())
-	if(SetEResoAndBias(dataSet->GetGEreso(),dataSet->GetGEbias()))
-	  cout << "Iact1dUnbinnedLkl::Iact1dUnbinnedLkl Warning: problems setting Ereso and Ebias graphs" << endl;
-      if(dataSet->GetMigMatrix()->GetEntries())
-	if(SetMigMatrix(dataSet->GetMigMatrix()))
-	  cout << "Iact1dUnbinnedLkl::Iact1dUnbinnedLkl Warning: problems setting MigMatrix histogram" << endl;   
-      if(dataSet->GetHdNdEpBkg()->GetEntries())
-	if(SetdNdEpBkg(dataSet->GetHdNdEpBkg()))
-	  cout << "Iact1dUnbinnedLkl::Iact1dUnbinnedLkl Warning: problems setting dNdEpBkg histogram" << endl;   
-      if(dataSet->GetHdNdEpFrg()->GetEntries())
-	if(SetdNdEpFrg(dataSet->GetHdNdEpFrg()))
-	  cout << "Iact1dUnbinnedLkl::Iact1dUnbinnedLkl Warning: problems setting dNdEpFrg histogram" << endl;   
+  for(Int_t i=0;i<fNon;i++)
+    {
+      dataSet->GetOnEntry(i);
+      fOnSample[i] = TMath::Log10(eventOnE);
+    }
+  for(Int_t i=0;i<fNoff;i++)
+    {
+      dataSet->GetOffEntry(i);
+      fOffSample[i] = TMath::Log10(eventOffE);
     }
   
+  // extract and copy IRFs
+  if(dataSet->GetHAeff())
```

Comment:
@javierrico, for all these histograms you were checking on the number of entries, e.g.
```C++
if(dataSet->GetHAeff()->GetEntries())
```
I removed the `GetEntries` because when I load the FITS files I do not create the empty histograms for the objects I am not going to use (e.g. `fHAeffOff`, `fHdNdEpBkg`). It is the same to check on the null pointer directly rather than on the entries of the histogram because each time you would eventually try to use one of this "secondaries" IRFs or counts, you will always check for their existence first.

Reply:
As we have discussed, this check cannot be removed because then empty histograms will be copied into Iact1dUnbinnedLkl class and then the class will try to use them (because it checks if the pointer is or not NULL and if it's not it assumes there are meaningful contents, which would be not the case if you remove the check)